utils.sh

```diff
@@ -389,6 +389,34 @@ EOF
     fi
 }
 
+function setup_release_mirror {
+
+    # combine global and local secrets
+    # pull from one registry and push to local one
+    # hence credentials are different
+
+    oc adm release mirror \
+       --insecure=true \
+        -a ${PULL_SECRET_FILE}  \
+        --from ${OPENSHIFT_RELEASE_IMAGE} \
+        --to-release-image ${LOCAL_REGISTRY_DNS_NAME}:${LOCAL_REGISTRY_PORT}/localimages/local-release-image:${OPENSHIFT_RELEASE_TAG} \
+        --to ${LOCAL_REGISTRY_DNS_NAME}:${LOCAL_REGISTRY_PORT}/localimages/local-release-image 2>&1 | tee ${MIRROR_LOG_FILE}
+    echo "export MIRRORED_RELEASE_IMAGE=$OPENSHIFT_RELEASE_IMAGE" > /tmp/mirrored_release_image
+
+    #To ensure that you use the correct images for the version of OpenShift Container Platform that you selected,
+    #you must extract the installation program from the mirrored content:
+    if [ -z "$KNI_INSTALL_FROM_GIT" ]; then
+      EXTRACT_DIR=$(mktemp --tmpdir -d "mirror-installer--XXXXXXXXXX")
+      _tmpfiles="$_tmpfiles $EXTRACT_DIR"
+
+      oc adm release extract --registry-config "${PULL_SECRET_FILE}" \
+        --command=openshift-baremetal-install --to "${EXTRACT_DIR}" \
+        "${LOCAL_REGISTRY_DNS_NAME}:${LOCAL_REGISTRY_PORT}/localimages/local-release-image:${OPENSHIFT_RELEASE_TAG}"
+
+      mv -f "${EXTRACT_DIR}/openshift-baremetal-install" ${OCP_DIR}
```

Comment:
I think that installer name should be eventually overridden, since in case of the agent flow should be `openshift-install`, while for the default (baremetal) flow remains `openshift-baremetal-install`. I also had a similar need previously, see https://github.com/openshift-metal3/dev-scripts/blob/1fe5e60b5587131e273149b96a07376f51eaeb41/03_build_installer.sh#L42, approached by defining the `OPENSHIFT_INSTALLER_CMD` (see https://github.com/openshift-metal3/dev-scripts/blob/1fe5e60b5587131e273149b96a07376f51eaeb41/agent/03_agent_build_installer.sh#L21).
Maybe now it could make sense to move it in https://github.com/openshift-metal3/dev-scripts/blob/master/agent/01_agent_requirements.sh?